src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareGenerateScriptOperation.cs

```diff
@@ -38,16 +38,23 @@ class SchemaCompareGenerateScriptOperation : ITaskOperation
 
         public SchemaComparisonResult ComparisonResult { get; set; }
 
+        public SchemaCompareScriptGenerationResult ScriptGenerationResult { get; set; }
+
         public SchemaCompareGenerateScriptOperation(SchemaCompareGenerateScriptParams parameters, SchemaComparisonResult comparisonResult)
         {
             Validate.IsNotNull("parameters", parameters);
-            Validate.IsNotNull("scriptFilePath", parameters.ScriptFilePath);
             this.Parameters = parameters;
             Validate.IsNotNull("comparisonResult", comparisonResult);
             this.ComparisonResult = comparisonResult;
         }
 
         public void Execute(TaskExecutionMode mode)
+        {
+            Execute();
+            AddScriptToTask();
+        }
+
+        public void Execute()
```

Comment:
It's strange to see two executes methods that are public. Any reason we can't have one public?

Reply:
This Execute() needs to be public because it's what the tests use. The other one Execute(TaskExecutionMode mode) needs to be public because that's what is being used by the service.

Reply:
Why can't the tests use Execute(TaskExecutionMode mode)?

Reply:
The AddScriptToTask() had to separated because the tests don't create a SqlTask, which is why Execute(TaskExecutionMode mode) calls Execute() then AddScriptToTask().

Reply:
Please write comments for issues like this.

Reply:
I put a comment about this above the AddScriptToTask() method

Reply:
I missed that comment. I think it's in the wrong spot to explain two public execute methods.

In the future, when you have to introduce workarounds like this, I'd try taking a step back and reconsider the design. For example, instead of taking on another service as a dependency, in this case SqlTask service, you could define and event ScriptCompleted. Then the service host could subscribe to the event and wire up the services. These keeps things decoupled.